### PR TITLE
Added test for Axes.bar_label

### PR DESCRIPTION
**Pull request by KheshavKumar**
## PR summary

I have added a datetime/timedelta smoke test for Axes.bar_label in lib/matplotlib/tests/test_datetime.py.
This addresses the Axes.bar_label task from https://github.com/matplotlib/matplotlib/issues/26864.

image

## PR checklist

- [N/A] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [N/A] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [N/A] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/document.html#write-examples-and-tutorials)
- [N/A] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/coding_guide.html#new-features-and-api-changes)
- [N/A] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

### CODE REVIEW

File: lib/matplotlib/tests/test_datetime.py [hunk 1]
```diff
@@ -140,11 +140,49 @@ def test_bar(self):
...
+        axs[1, 0].xaxis.set_major_formatter(mpl.dates.DateFormatter('%Y-%m-%d'))
+        axs[1, 0].bar_label(axs[1, 0].containers[0], fmt='%d',
+                             label_type='center', color='white')
+
+        # Variation 4: Label outside, with rotated text
+        axs[1, 1].bar(date_list, values)
+        axs[1, 1].xaxis_date()
+        axs[1, 1].xaxis.set_major_formatter(mpl.dates.DateFormatter('%Y-%m-%d'))
+        axs[1, 1].bar_label(axs[1, 1].containers[0], fmt='%d',
+                             label_type='edge', color='red', rotation=45)
```
Comment (jklymak) [inline lib/matplotlib/tests/test_datetime.py:179]: I don't think there is anything gained by testing all these variations.  I think all you want to test here is whether the x-positioning of the text works with a datetime axis.  

```suggestion
```

File: lib/matplotlib/tests/test_datetime.py [hunk 2]
```diff
@@ -140,11 +140,49 @@ def test_bar(self):
...
+                      datetime.timedelta(days=i) for i in range(5)]
+        values = [10, 20, 15, 25, 30]
+
+        # Create a bar plot
+        fig, axs = plt.subplots(2, 2, figsize=(10, 8))
+        fig.suptitle('Variations of ax.bar_label', fontsize=16)
+
+        # Variation 1: Default settings
+        axs[0, 0].bar(date_list, values)
+        axs[0, 0].xaxis_date()
```
[the remark this answers is not in the document]
Comment (jklymak) [reply lib/matplotlib/tests/test_datetime.py:156]: You don't need it or the DateFormatter.

**Review by jklymak — APPROVED** [review]
Looks good to me.  Thanks!

**Comment by ksunden** [comment]
#27033 is a test for `bar`, which also exercises e.g. the `width` and `bottom` args.


I think you can just add back the `bar_label` call and be good... I could see an argument for testing dates on each of the axes, but not so concerned with `bar_label` compared to a lot of methods, so we can start with just the simple case, I think.

File: lib/matplotlib/tests/test_datetime.py [hunk 3]
```diff
@@ -140,11 +140,18 @@ def test_bar(self):
...
+        # Generate some example data with dateTime inputs
+        date_list = [datetime.datetime(2023, 1, 1) +
+                     datetime.timedelta(days=i) for i in range(5)]
+        values = [10, 20, 15, 25, 30]
+
+        # Creating the plot
+        fig, axs = plt.subplots(1, 1, figsize=(10, 8), layout='constrained')
+        axs.bar(date_list, values)
+
+        plt.show()
```
Comment (ksunden) [inline lib/matplotlib/tests/test_datetime.py:154]:
```suggestion
```

The show is causing the test failure, as it is not interactive in CI.

Comment (KheshavKumar) [reply lib/matplotlib/tests/test_datetime.py]: Hi I have removed this line in the most recent commit

**Comment by KheshavKumar** [comment]
Apologies for the mistake! I added back a line with the bar_label function let me know how it looks now/ if there is anything else I should change

**Review by jklymak — APPROVED** [review]

File: lib/matplotlib/tests/test_datetime.py [hunk 4]
```diff
@@ -140,11 +140,20 @@ def test_bar(self):
...
+                     datetime.timedelta(days=i) for i in range(5)]
+        values = [10, 20, 15, 25, 30]
+
+        # Creating the plot
+        fig, axs = plt.subplots(1, 1, figsize=(10, 8), layout='constrained')
+        bars = axs.bar(date_list, values)
+
+        # Add labels to the bars using bar_label
+        axs.bar_label(bars, labels=[f'{val}%' for val in values],
+                       label_type='edge', color='black')
```
Comment (QuLogic) [inline lib/matplotlib/tests/test_datetime.py:156]: There is only one Axes object, so it doesn't need to be plural.
```suggestion
        fig, ax = plt.subplots(1, 1, figsize=(10, 8), layout='constrained')
        bars = ax.bar(date_list, values)

        # Add labels to the bars using bar_label
        ax.bar_label(bars, labels=[f'{val}%' for val in values],
                     label_type='edge', color='black')
```

**Comment by QuLogic** [comment]
Thanks @KheshavKumar! Congratulations on your first PR to Matplotlib :tada: We hope to hear from you again.

**Pull request merged**